client/capital/index.tsx

```diff
@@ -112,7 +112,7 @@ const getRowsData = ( loans: CapitalLoan[] ) =>
 		const data = {
 			paid_out_at: {
 				value: loan.paid_out_at,
-				display: clickable( dateI18n( 'M j, Y', loan.paid_out_at ) ),
+				display: clickable( formatUserDateTime( loan.paid_out_at ) ),
```

Comment:
Kudos for tackling this @mgascam 🙌.

As I've been auditing our timezone inconsistencies (as shared on P2 paJDYF-fQl-p2), I've noticed some places where we are incorrectly inferring the TZ of API-response timestamp strings.

For example, this capital loans list view datetime will not be inferred as a UTC datetime unless we explicitly pass this to the `datei18n` date parser – one way is to add a `Z` TZ identifier suffix or use `moment.utc().toISOString()` which is used elsewhere in this codebase.

```suggestion
				display: clickable(
					formatUserDateTime( loan.paid_out_at + 'Z' )
				),
```

The new `formatUserDateTime` shared function has helped to fix some of these (e.g. disputes list). I wonder if there is a suitable way we can abstract away the API response datetime value parsing as well?

Although, ideally, we should return the TZ within the timestamp string in the API response, which could simplify datetime handling on the frontend.

Reply:
Thanks for the review and your comments @Jinksi, much appreciated 😄 

> one way is to add a Z TZ identifier suffix or use moment.utc().toISOString() which is used elsewhere in this codebase.

I completely agree with you, and I reached the same conclusion while working on this task. For context, my focus has been on how dates are rendered to the merchant, aiming to avoid manipulating the "input" dates directly. My primary concern was ensuring that dates are displayed accurately and without errors.

However, I now realize this approach may have been flawed. I think it’s safe to assume that we are working with dates in the UTC timezone ("Z"), but as you pointed out, there are cases where this timezone information is missing. In my opinion, we should consistently convert raw date inputs to UTC (I'm using moment.utc for this) before formatting them. This would allow us to rely on dateI18n for consistent formatting of dates and times.

I’ve already made a few commits (e.g., [857dbf0](https://github.com/Automattic/woocommerce-payments/pull/9635/commits/857dbf015c1044529c13e58a8e478d00889f3fda)) to incorporate the conversion to UTC, but there are still a few more to go. Hopefully, I’ll be able to request a new review by tomorrow in my timezone.

> The new formatUserDateTime shared function has helped to fix some of these (e.g. disputes list). I wonder if there is a suitable way we can abstract away the API response datetime value parsing as well?

I’m not sure I have a great answer for this one yet, sorry. Perhaps we could update the `formatUserDateTime` function to handle the UTC conversion by default. This might reduce the repetition of `moment.utc` calls across the codebase. However, there is a potential complication: we are also working with dates in UNIX format (epoch). We’d need to account for these formats too, and I’m not sure yet how complex that might become.

Reply:
> we are also working with dates in UNIX format (epoch). We’d need to account for these formats too, and I’m not sure yet how complex that might become.

Yes, agree that an incremental approach here is suitable, and using `moment.utc` before passing in to `formatUserDateTime` is a robust and clear way to ensure incoming API response timestamps are interpreted as UTC.

Reply:
I implemented two utility functions using `moment.utc`: [formatDateTimeFromString](https://github.com/Automattic/woocommerce-payments/pull/9635/files#diff-87e7d575d3a841713f1f229aa40ddf0a14bc352d9c4b840529c415ee7b96f04aR22) and [formatDateTimeFromTimestamp](https://github.com/Automattic/woocommerce-payments/pull/9635/files#diff-87e7d575d3a841713f1f229aa40ddf0a14bc352d9c4b840529c415ee7b96f04aR30). I noticed there was too much repetition, so these functions aim to simplify and abstract this complexity for developers using them.
The only caveat is that developers must ensure they use the appropriate function for the given input format (string or timestamp).

Reply:
Closing this thread as resolved, but please feel free to re-open